### PR TITLE
Oba branch

### PR DESCRIPTION
**Pull request by obajide028**
- Abstract class implementation added, controllers updated and all test passed.
-  Review feature added 
- Review testing pending....

### CODE REVIEW

File: app/Policies/ReviewPolicy.php
Comment (Intuneteq) [inline app/Policies/ReviewPolicy.php:1]: clean up your code

**Review by Intuneteq — CHANGES_REQUESTED** [review]
Please make the following changes

File: app/Policies/ReviewPolicy.php
Comment (obajide028) [reply app/Policies/ReviewPolicy.php:1]: done

File: app/Http/Requests/UpdatecommunityRequest.php
[the remark this answers is not in the document]
Comment (obajide028) [reply app/Http/Requests/UpdatecommunityRequest.php:1]: done

File: app/Http/Requests/UpdateReviewRequest.php
[the remark this answers is not in the document]
Comment (obajide028) [reply app/Http/Requests/UpdateReviewRequest.php:1]: done

File: app/Http/Requests/StoreReviewRequest.php
[the remark this answers is not in the document]
Comment (obajide028) [reply app/Http/Requests/StoreReviewRequest.php:1]: done

File: app/Http/Controllers/ReviewController.php
[the remark this answers is not in the document]
Comment (obajide028) [reply app/Http/Controllers/ReviewController.php:1]: done

File: .env.testing
[the remark this answers is not in the document]
Comment (obajide028) [reply .env.testing:1]: done

**Comment by obajide028** [comment]
All changes made

**Review by Intuneteq — APPROVED** [review]

**Pull request merged**